src/sentry/rules/conditions/event_frequency.py

```diff
@@ -125,3 +127,73 @@ def query_hook(self, event, start, end, environment_id):
             environment_id=environment_id,
             use_cache=True,
         )[event.group_id]
+
+
+percent_intervals = {
+    "1m": ("one minute", timedelta(minutes=1)),
+    "5m": ("five minutes", timedelta(minutes=5)),
+    "10m": ("ten minutes", timedelta(minutes=10)),
+    "30m": ("30 minutes", timedelta(minutes=30)),
+    "1h": ("one hour", timedelta(minutes=60)),
+}
+
+
+class EventFrequencyPercentForm(EventFrequencyForm):
+    intervals = percent_intervals
+    interval = forms.ChoiceField(
+        choices=[
+            (key, label)
+            for key, (label, duration) in sorted(
+                percent_intervals.items(),
+                key=lambda key____label__duration: key____label__duration[1][1],
+            )
+        ]
+    )
+    value = forms.IntegerField(widget=forms.TextInput())
+
+
+class EventFrequencyPercentCondition(BaseEventFrequencyCondition):
+    label = "The issue has more errors than {value} percent of sessions in {interval}"
+
+    def __init__(self, *args, **kwargs):
+        self.intervals = percent_intervals
+        self.form_cls = EventFrequencyPercentForm
+        super().__init__(*args, **kwargs)
+
+    def query_hook(self, event, start, end, environment_id):
+        cache_key = f"r.c.spc:{environment_id}-{start.replace(minute=0, second=0, microsecond=0, tzinfo=None)}-{end.replace(minute=0, second=0, microsecond=0, tzinfo=None)}".replace(
```

Comment:
Sorry I'm new to this code, so I'm just trying to clarify:

This function gets run for every event that comes in. However you're caching the sessions results for one hour. Is start and end always 1 minute apart?

Reply:
Hey Evan. I'm just going to cache the session query for 5 minutes, that was the original decision that felt reasonable. The distance between start and end depend on the interval the user chooses for the rule and can be 1m, 5m, 10m, 30m, or 1h.

Reply:
OK that makes sense. For the record, given a time interval of 1m, the cache will refresh in less than 5 minutes as the interval straddles the hour. I don't think that's an issue though.

Reply:
Yeah definitely something good to be aware of, but I agree that it's not really an issue. Though, I am now feeling like I should almost write a comment about that edge case or just fix it somehow.